chrome/content/zotero/xpcom/data/feedItem.js

```diff
@@ -135,6 +140,18 @@ Zotero.FeedItem.prototype._saveData = Zotero.Promise.coroutine(function* (env) {
 	}
 });
 
+Zotero.FeedItem.prototype.toggleRead = Zotero.Promise.coroutine(function* (state) {
+	state = state !== undefined ? !!state : !this.isRead;
+	let changed = this.isRead != state;
+	this.isRead = state;
```

Comment:
I suppose it doesn't make much sense to update this if it hasn't changed. I think it would be appropriate to move this into the `if` statement.